### PR TITLE
feat(demo): add fragment links to doc

### PR DESCRIPTION
**Pull request by thbt**
Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
@thbt this needs a bit more work:
* CI build is failing on lint, needs fixing: https://travis-ci.org/ng-bootstrap/ng-bootstrap/builds/219690818
* there is quite a bit of code duplication, could you think of minimising it?

**Comment by thbt** [comment]
Hi @pkozlowski-opensource I made changes that should fix the build and reduce code duplication (I could reduce it further with inheritance if you want)

I also added an svg icon file distributed under CC-BY license ([link to project](https://dribbble.com/shots/1934932-77-Essential-Icons-Free-Download)), but I don't know where I should add the credits to the author, which file should I modify ?

File: demo/src/app/components/shared/api-docs/api-docs-class.component.ts [hunk 1]
```diff
@@ -19,13 +22,30 @@ import {Analytics} from '../../../shared/analytics/analytics';
...
+      display: inline-block;
+    }
     `
   ]
 })
-export class NgbdApiDocsClass {
+export class NgbdApiDocsClass implements AfterViewInit, OnDestroy {
   apiDocs: ClassDesc;
+  routeFragmentSubscription: Subscription;
+  scrollToFragment = scrollToFragment;
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/shared/api-docs/api-docs-class.component.ts]: Is this assignment necessary? Could you just call the imported function?

Comment (thbt) [inline demo/src/app/components/shared/api-docs/api-docs-class.component.ts]: The `click` event calling it was causing an error in the template file, but it was actually unneeded.

Comment (thbt) [inline demo/src/app/components/shared/api-docs/api-docs-class.component.ts]: For some reason clicking on the link was not working correctly (first click worked, then it changed the route but would not scroll, or it wouldnt scroll on init), I suppose moving the subscription to ngAfterViewInit fixed it

**Comment by wesleycho** [comment]
One way to get around the duplication is to create a component just for the `h2` element and its children. As far as I can tell, this can be successfully extracted into a component that has the appropriate inputs and outputs (output mainly for the ability to track clicks to the source code for the file).

**Pull request closed without merging**